core/src/main/scala/org/apache/spark/HttpServer.scala

```diff
@@ -160,7 +163,7 @@ private[spark] class HttpServer(
       throw new ServerStateException("Server is not started")
     } else {
       val scheme = if (securityManager.fileServerSSLOptions.enabled) "https" else "http"
-      s"$scheme://${Utils.localIpAddress}:$port"
+      s"$scheme://${InetAddresses.toUriString(InetAddress.getByName(Utils.localIpAddress))}:$port"
```

Comment:
This is a complex enough expression that it should just be a local val. Also, in the other file, the `java.net.InetAddress` import should sort above scala imports.